hs_core/hydroshare/utils.py

```diff
@@ -937,6 +937,9 @@ def add_file_to_resource(resource, f, folder='', source_name='',
     """
 
     # validate parameters
+    if resource.raccess.published:
+        raise ValidationError("Can't add files to a published resource")
+
```

Comment:
This check is already happening in the method that calls this util so it's happening twice.  I don't think access control checks belong in a util function.

Reply:
But there may be two paths to this function.

Reply:
That is true but access control checks still do not belong in a util function.  I realize our validation checks are leaked everywhere through the codebase and it's hard to make sense of it and know where to put things.  I don't know if I have the answer for this one but I don't think it should be here.  This utility function is called many times when files are uploaded.  It's for a single file, not a batch of files and is called multiple times when files are uploaded in a batch.